model_compression_toolkit/core/common/mixed_precision/resource_utilization_tools/resource_utilization_calculator.py

```diff
@@ -0,0 +1,646 @@
+              configurable weights.
+
+        Returns:
+            BOPS count.
+        """
+        node_mac = self.fw_impl.get_node_mac_operations(n, self.fw_info)
+        if node_mac == 0 or bitwidth_mode == BitwidthMode.Size:    # pragma: no cover
+            return node_mac
+
+        incoming_edges = self.graph.incoming_edges(n, sort_by_attr=EDGE_SINK_INDEX)
+        # TODO temporary adding this for const_representation test in torch which has Linear with const input
+        if not incoming_edges:
+            return 0
+        assert len(incoming_edges) == 1, \
+            f'Unexpected number of inputs {len(incoming_edges)} for BOPS calculation. Expected 1.'
+        input_act_node = incoming_edges[0].source_node
+        act_qc = act_qcs.get(input_act_node) if act_qcs else None
+        a_nbits = self._get_activation_nbits(input_act_node, bitwidth_mode, act_qc)
+
+        kernel_attrs = self.fw_info.get_kernel_op_attributes(n.type)
+        if len(kernel_attrs) > 1:
+            raise NotImplementedError('Multiple kernel attributes are not supported for BOPS computation.')
+        kernel_attr = kernel_attrs[0]
+        w_nbits = self._get_weight_nbits(n, kernel_attr, bitwidth_mode, w_qc)
+
+        node_bops = a_nbits * w_nbits * node_mac
+        return node_bops
+
+    @lru_cache
+    def _get_cut_target_nodes(self, cut: Cut, target_criterion: TargetInclusionCriterion) -> List[BaseNode]:
+        """
+        Retrieve target nodes from a cut filtered by a criterion.
+
+        Args:
+            cut: a graph cut.
+            target_criterion: criterion to include nodes for computation.
+
+        Returns:
+            A list of target nodes from a cut.
+        """
+        cut_nodes = [self.graph.find_node_by_name(e.node_name)[0] for e in cut.mem_elements.elements]
+        return self._get_target_activation_nodes(target_criterion, include_reused=True, nodes=cut_nodes)
+
+    def _get_target_weight_nodes(self,
+                                 target_criterion: TargetInclusionCriterion,
+                                 include_reused: bool) -> List[BaseNode]:
+        """
+        Collect nodes to include in weights utilization computation.
+
+        Args:
+            target_criterion: criterion to include weights for computation.
+            include_reused: whether to include reused nodes.
+
+        Returns:
+            Target nodes.
+        """
+        if target_criterion == TargetInclusionCriterion.QConfigurable:
+            nodes = self.graph.get_weights_configurable_nodes(self.fw_info, include_reused_nodes=include_reused)
+        elif target_criterion == TargetInclusionCriterion.AnyQuantized:
+            nodes = [n for n in self.graph if n.has_any_weight_attr_to_quantize()]
+        elif target_criterion == TargetInclusionCriterion.QNonConfigurable:
+            # TODO this is wrong. Need to look at specific weights and not the whole node
```

Comment:
What is wrong? (add details to the comment if you are not planning on taking care of it in this PR, so we understand what is the problem when we come to fix it)